src/GitHub.App/Services/GitClient.cs

```diff
@@ -136,6 +135,33 @@ public Task<Remote> GetHttpRemote(IRepository repo, string remote)
             });
         }
 
+        public async Task<string> ExtractFile(IRepository repository, string commitSha, string fileName)
+        {
+            var commit = repository.Lookup<Commit>(commitSha);
+            var blob = commit[fileName]?.Target as Blob;
+
+            var tempDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var tempFileName = $"{Path.GetFileNameWithoutExtension(fileName)}@{commitSha}{Path.GetExtension(fileName)}";
+            var tempFile = Path.Combine(tempDir, tempFileName);
+
+            Directory.CreateDirectory(tempDir);
+
+            if (blob != null)
+            {
+                using (var source = blob.GetContentStream(new FilteringOptions(fileName)))
+                using (var destination = File.OpenWrite(tempFile))
```

Comment:
As far as I'm aware this is the accepted way to chain `using` statements with the same scope; at least I've seen this pattern in every codebase I've ever looked at ;)

http://stackoverflow.com/a/9396092/6448
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/AES/AesCipherTests.cs#L466
https://github.com/github/Windows/blob/3f19e0a8496352208f85460ef01a17e2250b7d9b/GitHub/Helpers/ScreenshotHelper.cs#L21